code/go/0chain.net/smartcontract/storagesc/models.go

```diff
@@ -301,18 +349,28 @@ func (sng StorageNodeGeolocation) validate() error {
 	return nil
 }
 
+type partitionLocation struct {
+	Index      int              `json:"index"`
+	StartRound int64            `json:"start_round"`
+	Timestamp  common.Timestamp `json:"timestamp"`
+}
+
 // StorageNode represents Blobber configurations.
 type StorageNode struct {
-	ID              string                 `json:"id"`
-	BaseURL         string                 `json:"url"`
-	Geolocation     StorageNodeGeolocation `json:"geolocation"`
-	Terms           Terms                  `json:"terms"`    // terms
-	Capacity        int64                  `json:"capacity"` // total blobber capacity
-	Used            int64                  `json:"used"`     // allocated capacity
-	LastHealthCheck common.Timestamp       `json:"last_health_check"`
-	PublicKey       string                 `json:"-"`
+	ID                string                 `json:"id"`
+	BaseURL           string                 `json:"url"`
+	Geolocation       StorageNodeGeolocation `json:"geolocation"`
+	Terms             Terms                  `json:"terms"`             // terms
+	Capacity          int64                  `json:"capacity"`          // total blobber capacity
+	Used              int64                  `json:"used"`              // allocated capacity
+	BytesWritten      int64                  `json:"dataWritten"`       // in bytes
+	DataReadLR        float64                `json:"dataRead"`          // in GB
+	DRUpdateLastRound int64                  `json:"dataReadLastRound"` // last round when data read was updated
```

Comment:
Rename to DataReadUpdateLastRound if this is needed, json tag to: data_read_update_last_round